src/main.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright (c) 2022 by Delphix. All rights reserved.
+#
+
+"""
+This python script takes care of sending the data
+from DCT APIs for a set of components to New relic.
+It can be run using Python 3.8
+"""
+import datetime
+import os.path
+import sys
+import time
+
+import newrelic_telemetry_sdk.client
+import requests
+import logging
+import urllib3
+import configparser
+import tenacity
+
+from newrelic_telemetry_sdk import Event, EventClient
+from urllib3.exceptions import InsecureRequestWarning
+from src import VERSION
+
+urllib3.disable_warnings(InsecureRequestWarning)
+
+config_file = "dct_nr_config.ini"
+config = configparser.ConfigParser()
+
+if not os.path.exists(config_file):
+    raise Exception("Config file is required. "
+                    "Please add dct_nr_config.ini in the current directory")
+
+config.read_file(open("dct_nr_config.ini"))
+config_log_level = config['LOGGING']['LEVEL']
+INTERVAL = int(config['INTERVAL']['seconds'])
+
+formatter = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+logging_levels = {
+    'INFO': logging.INFO,
+    'DEBUG': logging.DEBUG,
+    'WARN': logging.WARNING,
+    'ERROR': logging.ERROR
+}
+logging.basicConfig(level=logging_levels[config_log_level], format=formatter)
+
+
+DCT_HOST_URL = os.environ.get('DCT_HOST_URL')
+DCT_API_KEY = os.environ.get('DCT_API_KEY')
+NEW_RELIC_INSERT_KEY = os.environ.get('NEW_RELIC_INSERT_KEY')
+
+if (DCT_HOST_URL and DCT_API_KEY and NEW_RELIC_INSERT_KEY) is None:
+    raise Exception("Please add the below keys as environment variables: "
+                    "\n1. DCT_HOST_URL"
+                    "\n2. DCT_API_KEY"
+                    "\n3. NEW_RELIC_INSERT_KEY")
+
+DCT_API_URL = f'https://{DCT_HOST_URL}/v2/'
+QUERY = '?limit=1000'
+DLPX_TYPES = [i.strip() for i in config['COMPONENTS']['monitor'].split(',')]
+
+# Request Headers ...
+req_headers = {
+    'Authorization': DCT_API_KEY  # noqa
+}
+
+# Python session, also handles the cookies ...
+session = requests.session()
+
+
+@tenacity.retry(
+    retry=tenacity.retry_if_exception_type(
+        newrelic_telemetry_sdk.client.HTTPError
+    ),
+    stop=tenacity.stop_after_attempt(5),
+    wait=tenacity.wait_fixed(5),
+    before_sleep=tenacity.before_sleep_log(logging, logging.INFO),
+)
+def send_event(event):
+    logging.debug(event)
+    event_client = EventClient(NEW_RELIC_INSERT_KEY)
+    response = event_client.send(event)
+    response.raise_for_status()
+    logging.debug("Event sent successfully!\n")
+
+
+@tenacity.retry(
+    retry=tenacity.retry_if_exception_type(
+        AssertionError
+    ),
+    stop=tenacity.stop_after_attempt(5),
+    wait=tenacity.wait_fixed(5),
+    before_sleep=tenacity.before_sleep_log(logging, logging.DEBUG),
+)
+def get_data_from_dct(component, query=QUERY):
+    url = DCT_API_URL + component + query
```

Comment:
Could we make this?

from urllib.parse import urljoin
urljoin(DCT_API_URL, component, query)

Reply:
Did not work out when I tried

```
from urllib.parse import urljoin
urljoin("https://agsdh.cdshj.sxs/v2/", "management/engines", "?limit=10")
'https://agsdh.cdshj.sxs/v2/management/engines'
```
Should have given
https://agsdh.cdshj.sxs/v2/management/engines?limit=10

Reply:
urljoin only considers 2 parameters to join, If needed I can use an "f string"

Reply:
`url = f"{DCT_API_URL}{component}{query}"`